File: app/src/main/java/me/imzack/app/end/injector/component/TypeEditComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.TypeEditPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.TypeEditActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeEditPresenterModule.class)
public interface TypeEditComponent {

    void inject(TypeEditActivity typeEditActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/TypeEditComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.TypeEditPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.TypeEditActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeEditPresenterModule::class))
interface TypeEditComponent {

    fun inject(typeEditActivity: TypeEditActivity)
}
